net/unbound/files/rootzone.sh

```diff
@@ -0,0 +1,130 @@
+# defaults and UCI scope variables define there. Function rootzone_update() is
+# intended to be part of a crontab or service stop to copy RFC5011 results
+# back to /etc/unbound/root.key.
+#
+##############################################################################
+
+rootzone_uci() {
+  # TODO: place holder; rootzone may be called out of context, so load few UCI
+  echo
+}
+
+##############################################################################
+
+roothints_update() {
+  # TODO: place holder; maybe we won't do this, but structure it in
+  echo
+}
+
+##############################################################################
+
+rootkey_update() {
+  if [ "$UNBOUND_N_ROOT_AGE" -lt 1 \
+      -o "$UNBOUND_N_ROOT_AGE" -gt 90 \
+      -o "$UNBOUND_B_DNSSEC" -lt 1 ] ; then
+    # Feature disabled
+    return 0
+  fi
+
+
+  state="$1"
+  drilltool=$( which drill )
+  digtool=$( which dig )
+  anchortool=$( which unbound-anchor )
+  
+  
+  if [ -f /etc/unbound/root.key ] ; then
+    basekey_date=$( date -r /etc/unbound/root.key +%s )
+    
+  else
+    # No persistent storage key
+    basekey_date=$( date -d 2000-01-01 +%s )
+  fi
+  
+  
+  if [ -f $UNBOUND_KEYFILE ] ; then
+    # Unbound maintains it itself
+    rootkey_date=$( date -r $UNBOUND_KEYFILE +%s )
+    rootkey_age=$(( (rootkey_date - basekey_date) / 86440 ))
+ 
+  elif [ "$state" = "stopping" ] ; then
+    # Service stop and not a cron job
+    rootkey_age=0
+    
+  elif [ -x "$anchortool" ] ; then
+    # No tmpfs key - use unbound-anchor
+    rootkey_date=$( date -I +%s )
+    rootkey_age=$(( (rootkey_date - basekey_date) / 86440 )) 
+    $anchortool -a $UNBOUND_KEYFILE
+  
+  elif [ -x "$drilltool" ] ; then
+    # No tmpfs key - use drill
+    rootkey_date=$( date -I +%s )
+    rootkey_age=$(( (rootkey_date - basekey_date) / 86440 )) 
+    $drilltool -D DNSKEY . > $UNBOUND_KEYFILE
+    
+  elif [ -x "$digtool" ] ; then
+    # No tmpfs key - use dig
+    rootkey_date=$( date -I +%s )
+    rootkey_age=$(( (rootkey_date - basekey_date) / 86440 )) 
+    $digtool +dnssec DNSKEY . > $UNBOUND_KEYFILE
+    
+  else 
+    # give up
+    rootkey_age=0
+  fi
+
+
+  if [ "$rootkey_age" -gt "$UNBOUND_N_ROOT_AGE" ] ; then
+    filestuff=$( cat $UNBOUND_KEYFILE )
+    
+      
+    case "$filestuff" in
+      *NOERROR*)
+        # Header comment for drill and dig
+        logger -t unbound -s "root.key updated after $rootkey_age days"
+        cp $UNBOUND_KEYFILE /etc/unbound/root.key
+        ;;
+        
+      *"state=2 [  VALID  ]"*)
+        # Comment inline to key for unbound-anchor
+        logger -t unbound -s "root.key updated after $rootkey_age days"
+        cp $UNBOUND_KEYFILE /etc/unbound/root.key
+        ;;
+        
+      *) 
+        logger -t unbound -s "root.key still $rootkey_age days old" 
+        ;;
+    esac
+  fi
+}
+
+##############################################################################
+
+rootzone_update() {
+  state="$1"
```

Comment:
Missing `local`, or do you mean to make it a global variable?

Applies to the other functions too.